src/pages/CookieClicker.vue

```diff
@@ -1,49 +1,331 @@
-]);
+  { name: 'Cursor', price: 15, cps: 0.1, count: 0 },
+  { name: 'Grandma', price: 100, cps: 1, count: 0 },
+  { name: 'Farm', price: 1000, cps: 10, count: 0 },
+  { name: 'Factory', price: 10000, cps: 100, count: 0 },
+])
+
+const achievements = ref([]) // Unlocked achievements
+const upgrades = ref([]) // Earned upgrades
 
-function buyBuilding(building){
-    cookies.value -= building.price;
-    building.price += Math.ceil(building.price / 100 * 15);
-    building.count++;
+// Buy a building
+function buyBuilding(building) {
+  cookies.value -= building.price
+  building.price += Math.ceil(building.price * 0.15)
+  building.count++
+  checkForUpgrades()
+  checkAchievements()
 }
+
+// Total cookies per second
 let cps = computed(() => {
-    let cps = 0;
-    buildings.value.forEach(building => {
-       cps+=building.cps*building.count;
-    });
-    return cps;
-});
-setInterval(()=>{
-   cookies.value+=cps.value;
-
-   document.title ='🍪' + +cookies.value.toFixed(1) + ' Cookies!';
-},1000);
+  let total = 0
+  buildings.value.forEach(b => total += b.cps * b.count)
+  if (upgrades.value.includes('Fast Fingers')) total *= 2
+  return total
+})
+
+// Cookie generation loop
+setInterval(() => {
+  cookies.value += cps.value * (goldenCookieBoost.value ? 10 : 1)
+
+  // Update page title
+  if (goldenCookieVisible.value) {
+    document.title = '🌟 GOLDEN COOKIE! 🍪'
+  } else if (goldenCookieBoost.value) {
+    document.title = '🚀 Boosted! 🍪 ' + cookies.value.toFixed(1)
+  } else {
+    document.title = '🍪 ' + cookies.value.toFixed(1) + ' Cookies!'
+  }
+}, 1000)
+
+// Random golden cookie spawns
+onMounted(() => {
+  setInterval(() => {
+    if (!goldenCookieVisible.value && Math.random() < 0.6) {
+      spawnGoldenCookie()
+    }
+  }, 15000)
+})
+
+// Show golden cookie at random position
+function spawnGoldenCookie() {
+  goldenCookieVisible.value = true
+  goldenCookiePosition.value = {
+    top: `${Math.floor(Math.random() * 80) + 5}%`,
+    left: `${Math.floor(Math.random() * 80) + 5}%`,
+  }
+
+  goldenCookieTimeout.value = setTimeout(() => {
+    goldenCookieVisible.value = false
+  }, 5000)
+}
+
+// Handle golden cookie click
+function clickGoldenCookie() {
+  const el = document.querySelector('.golden-cookie img')
+  if (el) {
+    el.classList.add('golden-clicked')
+    setTimeout(() => el.classList.remove('golden-clicked'), 800)
+  }
+
+  goldenCookieVisible.value = false
+  goldenCookieBoost.value = true
+  show("🌟 Golden Cookie activated! x10 CPS")
+  clearTimeout(goldenCookieTimeout.value)
+
+  setTimeout(() => {
+    goldenCookieBoost.value = false
+  }, 10000)
+}
+
+// Show toast message
+function show(message) {
+  toastMessage.value = message
+  showToast.value = true
+  setTimeout(() => {
+    showToast.value = false
+  }, 3000)
+}
+
+// Unlock achievements
+function checkAchievements() {
+  const unlocked = []
 
+  if (cookies.value >= 1000) unlocked.push("Cookie Enthusiast 🍪")
```

Comment:
Achievements are essentially bunch of if statements but you can come up with something nicer a bit. Try to come up with system where you could add achievments more easier. currently you will be repeating find and unlocked.push ALOT. Maybe try having achievments and in array of objects and you can have a check function in there as well.